Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java

```diff
@@ -59,37 +75,73 @@
      * @param req
      *            Holds HTTP request information
      * @return
+     * @throws IllegalAccessException
+     * @throws IllegalArgumentException
      */
     @RequestMapping(value = "rest/ConceptSearch", method = RequestMethod.GET, produces = "application/xml; charset=utf-8")
-    public @ResponseBody ResponseEntity<String> searchConcept(HttpServletRequest req) {
-        Map<String, String[]> queryParams = req.getParameterMap();
+    public @ResponseBody ResponseEntity<String> searchConcept(
+            @Validated ConceptSearchParameters conceptSearchParameters,
+            BindingResult result)
+            throws IllegalArgumentException, IllegalAccessException {
```

Comment:
In this method we are using reflection technique to fetch the searchFields value. So it throws these exceptions.